### ISSUE TITLE
Understand libehp and compare it to LIEF

### ISSUE THREAD

**Issue by standard3** [horde-re/gtirb-rs#9]
[libehp](https://github.com/horde-re/libehp) is a dependency of [ddisasm](https://github.com/GrammaTech/ddisasm/), the goal is to document it to know exactly what it does and if it can be replaced by Quarkslab [LIEF ](https://lief.re/)(which has Rust and Python bindings).

**Comment by standard3** [comment]
Thanks @ROYERidk, I really don't understand what you described, can you use more technical terms and more context ?

What do you mean by "used for error/exception handling (try/catch) while disassembling binary" ? 
The GTIRB structure is created and used by [ddisasm](https://github.com/GrammaTech/ddisasm/) once the binary has already been parsed. So are those errors handled while parsing the binary format ? If so, what are those exactly ?

LIEF is a powerful tool that allows to access every field on ELF/PE/Mach-O binary formats, what are we missing from that we can find in libehp ?

LIEF is already a dependency of ddisasm, that's why I would like to keep it simple stupid. See the dependency graph :

**Comment by ROYERidk** [comment]
@standard3 If i understand well, you want to know if we need to keep libehp and LIEF or if LIEF can do the job alone ?

**Comment by standard3** [comment]
@ROYERidk Yes that was the question. Ok that's more clear thanks, so we could potentially leave libehp behind.
